File: june14/july7ev_Rec5/pss.cpp
#include<iostream>
#include<string>

using namespace std;

void pss(string q,string a)
{
    if(q.length()==0)
    {
        cout << a << endl;
        return;
    }
    char ch=q[0];
    string roq=q.substr(1);
    pss(roq,a+"-");
    pss(roq,a+ch);
}
int main(int argc,char** argv)
{
    pss("abc","");
}

File: feb2/mar2_Rec4/pss.java
package mar2;

public class pss {
	
	public static void pss(String ques,String asf){
		if(ques.length()==0){
			System.out.println(asf);
			return;
		}
		char ch=ques.charAt(0);
		String roq=ques.substring(1);
		
		pss(roq, asf+ch);
		pss(roq, asf+'-');
	}

	public static void main(String[] args) {
		pss("abc","");

	}

}
